cmd/erasure-object.go

```diff
@@ -505,6 +505,10 @@ func renameData(ctx context.Context, disks []StorageAPI, srcBucket, srcEntry str
 
 	g := errgroup.WithNErrs(len(disks))
 
+	fvID := mustGetUUID()
+	for index := range disks {
+		metadata[index].SetTierFreeVersionID(fvID)
+	}
```

Comment:
I thought we shall add this information only conditionally such as if bucket is tiered etc. are we planning to save this for all I/O operations?

Reply:
> I thought we shall add this information only conditionally such as if bucket is tiered etc. are we planning to save this for all I/O operations?

We pass down a unique UUID to be used by xl-storage layer to create a free-version, if required. xl-storage layer determines if a free-version should be created by inspecting the version that is being modified/deleted.

tl;dr we pass down a unique UUID unconditionally, but a free-version is created only when needed.

Reply:
okay, that is not clear in the comments, that's why I got confused.